components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -18,77 +25,62 @@ import mozilla.components.feature.toolbar.internal.URLRenderer
 @Suppress("TooManyFunctions")
 class ToolbarPresenter(
     private val toolbar: Toolbar,
-    private val sessionManager: SessionManager,
-    private val sessionId: String? = null,
+    private val store: BrowserStore,
+    private val customTabId: String? = null,
     urlRenderConfiguration: ToolbarFeature.UrlRenderConfiguration? = null
-) : SelectionAwareSessionObserver(sessionManager) {
+) {
 
     @VisibleForTesting
     internal var renderer = URLRenderer(toolbar, urlRenderConfiguration)
 
+    private val scope = MainScope()
+
     /**
      * Start presenter: Display data in toolbar.
      */
     fun start() {
-        observeIdOrSelected(sessionId)
-        initializeView()
-
         renderer.start()
-    }
 
-    override fun stop() {
-        super.stop()
-
-        renderer.stop()
+        scope.launch {
+            store.channel()
+                .map { state -> state.findCustomTabOrSelectedTab(customTabId) }
+                .consumeEachIfChanged(
+                    then = { render(it) },
+                    otherwise = { clear() }
+                )
+        }
     }
 
     /**
-     * A new session has been selected: Update toolbar to display data of new session.
+     * Stop presenter: The toolbar will not be updated anymore.
      */
-    override fun onSessionSelected(session: Session) {
-        super.onSessionSelected(session)
-        initializeView()
-    }
-
-    override fun onAllSessionsRemoved() {
-        initializeView()
-    }
+    fun stop() {
+        scope.cancel()
```

Comment:
This will cancel the scope and its job and make it unusable, if I understand correctly. So, `onStart` wouldn't work anymore (we wouldn't be able to "launch" on the scope anymore). So far, we always had to create a new job because of this: https://github.com/mozilla-mobile/android-components/blob/master/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt#L59